store/sqlstore/file_info_store.go

```diff
@@ -44,6 +44,16 @@ type fileInfoWithChannelID struct {
 	Archived        bool
 }
 
+// Find the end of the term and include the wildcard if it exists
+// because we will put a wildcard at the end of the term regardless of whether a wildcard exists or not
+var wildcardRegExpForFileSearch = regexp.MustCompile(`\*?$`)
+
+// Find all characters between quotes
+var exactPhraseRegExpForFileSearch = regexp.MustCompile(`"[^"]+"`)
+
+// In excludedTerms case, we don't put a wildcard after the term.
+var wildcardRegExpForPostgresExcludedTerms = regexp.MustCompile(`\*($| )`)
+
```

Comment:
And some more rewording here, although I must admit I don't fully understand this regex.
```suggestion
// Regexp to match an asterisk, either at the end of the string, or right before a space.
// It is used in the file search logic so that we don't put a wildcard after the excluded terms.
var wildcardRegExpForPostgresExcludedTerms = regexp.MustCompile(`\*($| )`)

```